Error on Transferring Archetype

```
panic: runtime error: index out of range [-5]

goroutine 69 [running]:
github.com/yohamta/donburi/internal/storage.(*Archetype).SwapRemove(...)
```

--- a/internal/storage/components.go
+++ b/internal/storage/components.go
@@ -39,9 +39,7 @@ func (cs *Components) PushComponents(components []*component.ComponentType, arch
 }
 
 // MoveComponent moves the pointer to data of the component in the archetype.
-func (cs *Components) MoveComponent(c *component.ComponentType, src ArchetypeIndex, idx ComponentIndex, dst ArchetypeIndex) {
-	storage := cs.Storage(c)
-	storage.MoveComponent(src, idx, dst)
+func (cs *Components) Move(src ArchetypeIndex, dst ArchetypeIndex) {
 	cs.componentIndices[src]--
 	cs.componentIndices[dst]++
 }

--- a/internal/storage/components_test.go
+++ b/internal/storage/components_test.go
@@ -58,7 +58,8 @@ func TestComponents(t *testing.T) {
 	var srcArchIdx ArchetypeIndex = target.archIdx
 	var dstArchIdx ArchetypeIndex = 1
 
-	components.MoveComponent(ca, srcArchIdx, target.compIdx, dstArchIdx)
+	storage.MoveComponent(srcArchIdx, target.compIdx, dstArchIdx)
+	components.Move(srcArchIdx, dstArchIdx)
 
 	if storage.Contains(srcArchIdx, target.compIdx) {
 		t.Errorf("storage should not contain the component at %d, %d", target.archIdx, target.compIdx)

--- a/world.go
+++ b/world.go
@@ -194,12 +194,14 @@ func (w *world) TransferArchetype(from, to storage.ArchetypeIndex, idx storage.C
 
 	// move components
 	for _, component_type := range from_layout.Components() {
+		storage := w.components.Storage(component_type)
 		if to_layout.HasComponent(component_type) {
-			w.components.MoveComponent(component_type, from, idx, to)
+			storage.MoveComponent(from, idx, to)
 		} else {
-			w.components.Remove(from_arch, idx)
+			storage.SwapRemove(from, idx)
 		}
 	}
+	w.components.Move(from, to)
 
 	return storage.ComponentIndex(len(to_arch.Entities()) - 1)
 }

